### PR TITLE
Revert "fix(cdn-logs-report): stop emitting citability score + deployed-at-edge (LLMO-5646)"

### PR DESCRIPTION
**Pull request by akshaymagapu**
## What

Reverts #2633 (merge commit `910ad1a`), which removed server-side citability derivation from `agentic-traffic-mapper.js`.

## Why

Restores emission of `citability_score` and `deployed_at_edge` into the agentic-traffic `dimensions` JSONB from the daily export.

## Restored behavior

In `src/cdn-logs-report/utils/agentic-traffic-mapper.js`:
- `buildCitabilityMap()` and `getCitabilityScores()` helpers
- Per-URL citability lookup in `mapToAgenticTrafficBundle`, re-populating `dimensions.citability_score` and `dimensions.deployed_at_edge`

The restored code sources scores from `context.dataAccess.PageCitability` and is fully self-contained — no dangling references to any file removed in #2633.

## Validation

- `test/audits/cdn-logs-report/**` — 184 passing
- `eslint` — clean on the changed source file

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/cdn-logs-report/utils/agentic-traffic-mapper.js [hunk 1]
```diff
@@ -124,8 +163,17 @@ export async function mapToAgenticTrafficBundle(rows, site, context, trafficDate
...
 
       const host = normalizeText(row.host, defaultHost) || defaultHost;
+      const citability = citabilityMap[urlPath];
       const dimensions = {};
 
+      if (citability?.score !== undefined && citability?.score !== null) {
+        dimensions.citability_score = citability.score;
+      }
+
+      if (citability?.isDeployedAtEdge !== undefined) {
```
Comment (MysticatBot) [inline src/cdn-logs-report/utils/agentic-traffic-mapper.js]: issue (blocking): The null guard on `deployed_at_edge` is asymmetric with `citability_score`.

```javascript
// citability_score excludes both null and undefined:
if (citability?.score !== undefined && citability?.score !== null)

// deployed_at_edge excludes only undefined:
if (citability?.isDeployedAtEdge !== undefined)
```

If `getIsDeployedAtEdge()` returns `null` (e.g. a record where the field has not been computed yet), this will emit `deployed_at_edge: null` into the dimensions JSONB. The score guard correctly omits the field in that case. This inconsistency can push null values into downstream analytics.

Fix: add `&& citability?.isDeployedAtEdge !== null` to the guard, or if emitting null is intentional, add the same treatment to `citability_score` for consistency. Either way, add a test case with `getIsDeployedAtEdge: () => null` to document the intended behavior.

File: test/audits/cdn-logs-report/agentic-daily-export.test.js [hunk 2]
```diff
@@ -304,7 +304,6 @@ describe('agentic daily export', () => {
 
     expect(module.testHelpers.escapeCsvValue(null)).to.equal('');
     expect(module.testHelpers.escapeCsvValue(undefined)).to.equal('');
```
Comment (MysticatBot) [inline test/audits/cdn-logs-report/agentic-daily-export.test.js:306]: issue (blocking): This assertion was removed but is unrelated to the citability revert:

```javascript
expect(module.testHelpers.escapeCsvValue('a,"b"')).to.equal('"a,""b"""');
```

This tested the quoting/escaping branch of `escapeCsvValue` that handles strings containing commas and quotes together. The implementation is unchanged, so this branch is now untested. Given the repo's 100% coverage requirement (CLAUDE.md: "Coverage enforcement is strict: 100% lines, branches, and statements across all src/**/*.js"), this may fail CI.

Fix: restore this assertion. It is a one-liner orthogonal to the revert.

**Review by MysticatBot — CHANGES_REQUESTED** [review]
Hey @akshaymagapu,

> ⚠ Degraded review - no spec document was found for this change (searched the PR links, the touched repos' docs, the architecture/guidelines docs, and linked Jira). This review covers code-level quality but could not validate the change against an agreed design, so confidence is reduced. Add a spec link (PR template section 4) and re-request review for a full-confidence pass.

**Verdict: Request changes** - two issues to resolve before merge, both straightforward.
**Complexity: MEDIUM** - medium-sized revert with test refactoring, single service.
_Changes: Reverts removal of server-side citability score and deployed-at-edge emission in the agentic traffic mapper, restoring dimensions JSONB population from PageCitability records (3 files)._

Note: CI checks are currently pending - verify before merge.

## Must fix before merge

1. [Important] Asymmetric null guard on `deployed_at_edge` will emit null values - `src/cdn-logs-report/utils/agentic-traffic-mapper.js:173` (details inline)
2. [Important] Removed `escapeCsvValue` test assertion reduces coverage of a production code path unrelated to this revert - `test/audits/cdn-logs-report/agentic-daily-export.test.js:306` (details inline)

<details>
<summary>Non-blocking (2): minor issues and suggestions</summary>

- nit: No test for `citability_score: 0` (falsy but valid value that passes the null guard) - `test/audits/cdn-logs-report/agentic-traffic-mapper.test.js`
- suggestion: Store epoch millis in the citability map accumulator to avoid repeated `new Date()` construction during the reduce comparison - `src/cdn-logs-report/utils/agentic-traffic-mapper.js:93`
</details>

---

<sub><b>Skill:</b> pr-review | <b>Model:</b> us.anthropic.claude-opus-4-6-v1[1m] | <b>Duration:</b> 2m 54s | <b>Cost:</b> $5.83 | <b>Commit:</b> <code>2b3a0bf245e0a1b2ea4fe0d1c97ec4680b7f3a81</code><br>If this code review was useful, please react with 👍. Otherwise, react with 👎.</sub>

**Comment by akshaymagapu** [comment]
Thanks for the review. This PR is a **pure `git revert` of #2633** — the intent is to restore `main` to its exact pre-#2633 state, not to introduce new behavior. With that framing, pushing back on both blocking items:

**1. Asymmetric null guard on `deployed_at_edge` (mapper:173)** — This is the exact code that ran on `main` before #2633; the asymmetry is pre-existing and shipped in prod, so it's not a regression introduced here. #2633's own description confirms the downstream path is null-safe (`agentic_traffic_is_deployed_at_edge` → `false`, no error). "Fixing" the guard inside a revert would make the revert non-faithful and harder to reason about / re-revert. If we want to tighten the guard (and/or add a `getIsDeployedAtEdge: () => null` case), I'm happy to do it as a **separate follow-up** rather than smuggle it into the revert.

**2. Removed `escapeCsvValue('a,"b"')` assertion (daily-export.test:306)** — This assertion was **added by #2633**, so a faithful revert removes it. The concern was that it may drop below the 100% coverage gate — I verified that's not the case:

- `npx c8 --include 'src/cdn-logs-report/agentic-daily-export.js' --check-coverage --branches 100 --lines 100 --statements 100` → **100% stmts / branches / funcs / lines** for that file. The quoting branch (`/["\n,]/`) is covered by other tests exercising `serializeCsv`.
- CI `ci / build` is **green** and `codecov/patch` passes ("All modified and coverable lines are covered by tests").

So there's no CI/coverage risk, and keeping the assertion would deviate from the revert.

The two non-blocking nits (`citability_score: 0` test, epoch-millis perf) fall under the same rationale — both concern the restored pre-#2633 code, so I'd keep the revert faithful and address them separately if we decide they're worth it.

**Comment by akshaymagapu** [comment]
Addressed both blocking findings in `9064ff91`:

1. **Null guard** — `deployed_at_edge` guard is now symmetric with `citability_score` (also excludes `null`), so a `null` `getIsDeployedAtEdge()` no longer emits `deployed_at_edge: null`. Added a mapper test with `getCitabilityScore: () => null` + `getIsDeployedAtEdge: () => null` asserting both are omitted.
2. **CSV assertion** — restored `escapeCsvValue('a,"b"')` → `'"a,""b"""'`.

Validation: `test/audits/cdn-logs-report/**` 185 passing; `agentic-traffic-mapper.js` and `agentic-daily-export.js` both at 100% stmts/branches/funcs/lines; eslint clean.

**Review by cwjwisse — APPROVED** [review]

**Review by MysticatBot — APPROVED** [review]
Hey @akshaymagapu,

> ⚠ Degraded review - no spec document was found for this change (searched the PR links, the touched repos' docs, the architecture/guidelines docs, and linked Jira). This review covers code-level quality but could not validate the change against an agreed design, so confidence is reduced. Add a spec link (PR template section 4) and re-request review for a full-confidence pass.

**Verdict: Approve** - both prior blocking findings addressed correctly; fix commit is minimal and well-tested.
**Complexity: HIGH** - large diff (570 lines, test restructuring inflates count), single service.
_Changes: Reverts removal of server-side citability score and deployed-at-edge emission in the agentic traffic mapper, with a fix for the asymmetric null guard and restored test coverage (3 files)._

Note: CI build checks are currently pending - verify before merge.

## Previously flagged, now resolved

- Asymmetric null guard on `deployed_at_edge` - now symmetric with `citability_score` (excludes both undefined and null)
- Removed `escapeCsvValue('a,"b"')` assertion - restored in the fix commit
- New test added verifying both null score and null deployed-at-edge produce empty dimensions

---

<sub><b>Skill:</b> pr-review | <b>Model:</b> us.anthropic.claude-opus-4-6-v1[1m] | <b>Duration:</b> 6m 27s | <b>Cost:</b> $1.84 | <b>Commit:</b> <code>9064ff918522416c36f3cf6cc08ebe9325909112</code><br>If this code review was useful, please react with 👍. Otherwise, react with 👎.</sub>
